bindings/python/setup.py

```diff
@@ -0,0 +1,35 @@
+from setuptools import find_packages, setup, Extension
+from Cython.Build import cythonize
+
+import os
+
+
+LIBBSON_PATH = os.path.join(
+    os.path.dirname(os.path.abspath(__file__)), 'pymongoarrow/libbson/lib')
+
+
+EXT_MODULES = [
+    Extension('pymongoarrow.bson_util',
+              sources=['pymongoarrow/bson_util.pyx'],
+              language='c++',
+              include_dirs=['pymongoarrow/libbson/include'],
+              libraries=['bson-1.0'],
+              library_dirs=[LIBBSON_PATH],
+              extra_link_args=['-Wl,-rpath,' + LIBBSON_PATH])
```

Comment:
Are library_dirs and extra_link_args needed at all here? The docs suggest that only `libraries` is required for cython to find the library: https://cython.readthedocs.io/en/stable/src/tutorial/clibraries.html#dynamic-linking

> Copy libbson-1.0.0.<extension> to mongo-arrow/bindings/python/pymongoarrow/libbson/lib

I think if we remove library_dirs and extra_link_args then we can remove this requirement to copy libbson into the mongo-arrow dir.

Again from the cython dynamic-linking docs:

> If the libcalg is not installed in a ‘normal’ location, users can provide the required parameters externally by passing appropriate C compiler flags, such as:
  ```
  CFLAGS="-I/usr/local/otherdir/calg/include"  \
LDFLAGS="-L/usr/local/otherdir/calg/lib"     \
    python setup.py build_ext -i
  ```
  
So if you prefer not to install libbson on your system you could do:
  ```
  CFLAGS="-I/usr/local/otherdir/libbson/include"  \
LDFLAGS="-L/usr/local/otherdir/libbson/lib"     \
    python setup.py build_ext -i
  ```

Reply:
Thanks for this suggestion. This is how things are setup now. I have updated the PR description to reflect this change.

Reply:
Nice!